File: hoppity.c
#include <stdio.h>
#include <libgen.h>

//#define DEBUG

int main (int argc, char *argv[]) {

    if (argc <= 1) {
        fprintf(stderr, "You need to specify a filename!\n");
        fprintf(stderr, "Usage: %s <filename>\n", basename(argv[0]));
        return 1;
    }

#ifdef DEBUG
    printf("Trying to open %s ...\n", argv[1]);
#endif

    FILE *in_file = fopen(argv[1], "r");

    if (NULL == in_file) {
        fprintf(stderr, "That file doesn't exist or it's not readable!\n");
        return 1;
    }

    // remember, the number could be 0 or we could've got an error
    // atoi() doesn't distinguish or return NULL
    int to_count = -1;
    char buffer[100];

    do {
        fgets(buffer, 100, in_file);
        to_count = atoi(buffer);
    }
    while (!feof(in_file) && to_count <= 0);
    
    fclose(in_file);
   
#ifdef DEBUG
    printf("Got value of %d from file.\n", to_count);
#endif

    int i;

    for (i = 1; i <= to_count; ++i) {
        if (0 == i % 15) {
            printf("Hop\n");
        }
        else if (0 == i % 3) {
            printf("Hoppity\n");
        }
        else if (0 == i % 5) {
            printf("Hophop\n");
        }
    }

    return 0;

}

File: hoppity.php
<?php

// theoretically you should check
// if ('cli' === php_sapi_name())
// but I'm tired and you get what you pay for, ;)

define('DEBUG', false);

if (!isset($argv[1])) {
    @file_put_contents('php://stderr', "You need to specify a filename!\n" . "Usage: " . basename($_SERVER['PHP_SELF']) . " <filename>\n", FILE_APPEND);
    exit(1);
}

if (!file_exists($argv[1])) {
    @file_put_contents('php://stderr', "That file doesn't exist!\n", FILE_APPEND);
    exit(1);
}
if (!is_readable($argv[1])) {
    @file_put_contents('php://stderr', "That file isn't readable!\n", FILE_APPEND);
    exit(1);
}

$to_count = rtrim(file_get_contents($argv[1]));

if (DEBUG) {
    echo "Got value of $to_count from file.\n";
}

for ($i = 1; $i <= $to_count; ++$i) {
    if (0 == $i % 15) {
        echo "Hop\n";
    }
    elseif (0 == $i % 3) {
        echo "Hoppity\n";
    }
    elseif (0 == $i % 5) {
        echo "Hophop\n";
    }
}

?>
